crates/core/src/operations/cast.rs

```diff
@@ -1,36 +1,106 @@
 //! Provide common cast functionality for callers
 //!
-use arrow_array::{Array, ArrayRef, RecordBatch, StructArray};
+use arrow::datatypes::DataType::Dictionary;
+use arrow_array::{new_null_array, Array, ArrayRef, RecordBatch, StructArray};
 use arrow_cast::{cast_with_options, CastOptions};
-use arrow_schema::{DataType, Fields, SchemaRef as ArrowSchemaRef};
-
+use arrow_schema::{
+    ArrowError, DataType, Field as ArrowField, Fields, Schema as ArrowSchema,
+    SchemaRef as ArrowSchemaRef,
+};
 use std::sync::Arc;
 
 use crate::DeltaResult;
 
+pub(crate) fn merge_field(left: &ArrowField, right: &ArrowField) -> Result<ArrowField, ArrowError> {
+    if let Dictionary(_, value_type) = right.data_type() {
+        if value_type.equals_datatype(left.data_type()) {
+            return Ok(left.clone());
```

Comment:
That clone is not necessary I think

Reply:
I'll try getting rid of it, but will require explicit Lifetimes, I'll try

Reply:
Ok, I gave up. the problem is that try_merge of arrow requires a &mut to be passed, so a clone is required there anyway. and then you create a reference to something owned in the function... the alternative would be passing around Arc's, but that also is not nice since right is a reference anyway. Also we use it to create a schema instance which required the data to be owned anyway, so cloning is required there, too.